packages/common/test-builder/src/builder.ts

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { Peer } from './peer';
+
+type CreatePeerOpts<T extends Peer> = {
+  factory: () => T;
+};
+
+export class Builder {
+  private readonly _peers = new Set<Peer>();
+
+  createPeer<T extends Peer>(opts: CreatePeerOpts<T>): T {
```

Comment:
There isn't really enough context in this peer to assess if this is right -- there should always be minimally a test use case.

Reply:
> Also how does this relate to the existing dxos/client/testing builders?

Those builders should `extend` this one